### PR TITLE
chore: added dsa resource

### LINKED ISSUE #608: [Add] MIT 6.006 Introduction to Algorithms
### Site URL

https://youtube.com/playlist?list=PLUl4u3cNGP61Oq3tWYp6V_F-5jb5L2iHb

### Description of the above provided site

Explanation of Algorithms concepts. The resources for Algorithm are limited but this course is one of the best Algorithm resource available on Web.
Moreover, this course will help beginners to understand the topic of algorithm. This course contains video tutorials curated by Top MIT faculties. 
This would definitely be a good addition to the resources.

### Category

Youtube

### Sub Category

Algorithms

### Checklist

- [X] I have checked the existing issues
- [X] I have read the [Contributing Guidelines](https://github.com/rupali-codes/LinksHub/blob/main/CONTRIBUTING.md)
- [X] I am willing to work on this issue (optional)
- [X] I am a GSSoC'23 contributor

### PR DESCRIPTION
**Pull request by Akshatbhardwaj15**
## Fixes Issue

Closes  #608 
  
## Changes proposed

Added Algorithm tutorial in data structures section of Database.

##Screenshot
image

## Note to reviewers
Please review my change and add it to your main branch

### CODE REVIEW

**Comment by CBID2** [comment]
Hey @Akshatbhardwaj1! :) I closed this pull request because it's missing the issue number that it closes. Please add this information per the [CONTRIBUTING.MD](https://github.com/rupali-codes/LinksHub/blob/main/CONTRIBUTING.md) and then I’ll reopen your PR.

File: database/youtube/data-structures.json [hunk 1]
```diff
@@ -62,5 +62,13 @@
     "category": "youtube",
     "subcategory": "dsa",
     "language": "english"
+  },
+  {
+    "name": "MIT OpenCourseWare",
+    "description": "Explanation of Algorithms concepts. The resources for Algorithm are limited but this course is one of the best Algorithm resource available on Web.",
```
Comment (CBID2) [inline database/youtube/data-structures.json:68]: [Typo]
```suggestion
    "description": "Explanation of Algorithms concepts. The resources for Algorithm are limited but this course is one of the best Algorithm resources available on the Web.",
```

**Review by CBID2 — CHANGES_REQUESTED** [review]
Hi @Akshatbhardwaj15! 😊 I mentioned a typo to fix in my suggestion, so just work on that and you're ready to go! :)

**Comment by Akshatbhardwaj15** [comment]
i have fixed the typo, pls review it.

**Review by CBID2 — APPROVED** [review]
Well done @Akshatbhardwaj15! 😊 LGTM! 😊

**Pull request merged**